yarn-project/ivc-integration/src/client_ivc_integration.test.ts

```diff
@@ -0,0 +1,151 @@
+  MockPrivateKernelTailCircuit,
+  witnessGenCreatorAppMockCircuit,
+  witnessGenMockPrivateKernelInitCircuit,
+  witnessGenMockPrivateKernelInnerCircuit,
+  witnessGenMockPrivateKernelResetCircuit,
+  witnessGenMockPrivateKernelTailCircuit,
+  witnessGenReaderAppMockCircuit,
+} from './index.js';
+
+/* eslint-disable camelcase */
+
+const logger = createDebugLogger('aztec:clientivc-integration');
+
+jest.setTimeout(60_000);
+
+describe('Client IVC Integration', () => {
+  let bbWorkingDirectory: string;
+  let bbBinaryPath: string;
+
+  beforeEach(async () => {
+    // Create a temp working dir
+    bbWorkingDirectory = await fs.mkdtemp(path.join(os.tmpdir(), 'bb-client-ivc-integration-'));
+    bbBinaryPath = path.join(path.dirname(fileURLToPath(import.meta.url)), '../../../barretenberg/cpp/build/bin', 'bb');
+  });
+
+  async function createClientIvcProof(witnessStack: Uint8Array[], bytecodes: string[]): Promise<ClientIvcProof> {
+    await fs.writeFile(
+      path.join(bbWorkingDirectory, 'acir.msgpack'),
+      encode(bytecodes.map(bytecode => Buffer.from(bytecode, 'base64'))),
+    );
+
+    await fs.writeFile(path.join(bbWorkingDirectory, 'witnesses.msgpack'), encode(witnessStack));
+    const provingResult = await executeBbClientIvcProof(
+      bbBinaryPath,
+      bbWorkingDirectory,
+      path.join(bbWorkingDirectory, 'acir.msgpack'),
+      path.join(bbWorkingDirectory, 'witnesses.msgpack'),
+      logger.info,
+    );
+
+    if (provingResult.status === BB_RESULT.FAILURE) {
+      throw new Error(provingResult.reason);
+    }
+
+    return ClientIvcProof.readFromOutputDirectory(bbWorkingDirectory);
+  }
+
+  it('Should generate a verifiable client IVC proof from a simple mock tx', async () => {
+    const tx = {
+      number_of_calls: '0x1',
+    };
+    // Witness gen app and kernels
+    const appWitnessGenResult = await witnessGenCreatorAppMockCircuit({ commitments_to_create: ['0x1', '0x2'] });
+
+    const initWitnessGenResult = await witnessGenMockPrivateKernelInitCircuit({
+      app_inputs: appWitnessGenResult.publicInputs,
+      tx,
+    });
+
+    const tailWitnessGenResult = await witnessGenMockPrivateKernelTailCircuit({
+      prev_kernel_public_inputs: initWitnessGenResult.publicInputs,
+    });
+    // Create client IVC proof
+    const bytecodes = [
+      MockAppCreatorCircuit.bytecode,
+      MockPrivateKernelInitCircuit.bytecode,
+      MockPrivateKernelTailCircuit.bytecode,
+    ];
+    const witnessStack = [appWitnessGenResult.witness, initWitnessGenResult.witness, tailWitnessGenResult.witness];
+
+    const proof = await createClientIvcProof(witnessStack, bytecodes);
+    await proof.writeToOutputDirectory(bbWorkingDirectory);
+    const verifyResult = await verifyClientIvcProof(bbBinaryPath, bbWorkingDirectory, logger.info);
+
+    expect(verifyResult.status).toEqual(BB_RESULT.SUCCESS);
+  });
+
+  it('Should generate a verifiable client IVC proof from a complex mock tx', async () => {
```

Comment:
Both of these tests are quite readable but I also wouldn't mind a few overview comments here :)